File: rb/features.rb
# Api2newsEndpoint SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module Api2newsEndpointFeatures
  def self.make_feature(name)
    case name
    when "base"
      Api2newsEndpointBaseFeature.new
    when "test"
      Api2newsEndpointTestFeature.new
    else
      Api2newsEndpointBaseFeature.new
    end
  end
end

File: py/features.py
# Api2newsEndpoint SDK feature factory

from feature.base_feature import Api2newsEndpointBaseFeature
from feature.test_feature import Api2newsEndpointTestFeature


def _make_feature(name):
    features = {
        "base": lambda: Api2newsEndpointBaseFeature(),
        "test": lambda: Api2newsEndpointTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
